src/Storages/MergeTree/MergeTreeDataSelectExecutor.cpp

```diff
@@ -1704,8 +1707,24 @@ MarkRanges MergeTreeDataSelectExecutor::filterMarksUsingIndex(
 
                 if (index_helper->isVectorSimilarityIndex())
                 {
-                    auto rows = condition->calculateApproximateNearestNeighbors(granule);
-
+                    read_hints.ann_search_results = condition->calculateApproximateNearestNeighbors(granule);
+
+                    /// corresponding ranges have to be returned in ascending order
```

Comment:
Could you please move l. 1712 - l. 1720 back into MergeTreeIndexVectorSimilarity.cpp? The code does not belong here.